src/twisted/web/http.py

```diff
@@ -377,6 +379,75 @@ def parseContentRange(header):
 
 
 
+class _IDeprecatedHTTPChannelToRequestInterface(Interface):
+    """
+    The interface L{HTTPChannel} expects of L{Request}.
```

Comment:
hm, actually i'm loathe to put the reference from `HTTPChannel`'s docstring back to this interface, since it's supposed to be private.  if people subclass HTTPChannel i don't want them to feel like they should also construct a provider of this interface.